dart/lib/src/protocol/sentry_span.dart

```diff
@@ -28,30 +28,39 @@ class SentrySpan extends ISentrySpan {
     this._tracer,
     this._context,
     this._hub, {
+    DateTime? startTimestamp,
     bool? sampled,
     Function()? finishedCallback,
   }) {
+    _startTimestamp = startTimestamp?.toUtc() ?? getUtcDateTime();
     this.sampled = sampled;
     _finishedCallback = finishedCallback;
   }
 
   @override
-  Future<void> finish({SpanStatus? status}) async {
+  Future<void> finish({SpanStatus? status, DateTime? endTimestamp}) async {
     if (finished) {
       return;
     }
 
     if (status != null) {
       _status = status;
     }
-    _timestamp = getUtcDateTime();
+
+    // if endTimestamp is before startTimestamp, set throwable
+    if (endTimestamp?.isBefore(_startTimestamp) ?? false) {
+      throwable = StateError('endTimestamp cannot be before startTimestamp');
```

Comment:
This throwable exists to help us connect the transaction with the capture of the trowable. So in Sentry you can see in the issue details page the information about a transaction. And from the transaction, you can see an error occurred  and navigate to the issue detail. Here we better log out the issue so when the SDK is in debug mode we can see what's going on:


Whatever equivalent we have here:

```suggestion
      this._logger.logError('endTimestamp cannot be before startTimestamp');
```

Reply:
I believe `sentry_span` doesn't have access to the logger, but yeah, logging would be the correct solution.

Reply:
@ueman is right. `sentry_span` has no access to the logger. 
Is it acceptable to implement `_options`'s getter on the [Hub.dart](https://github.com/getsentry/sentry-dart/blob/4eb2218eefb9bf74ee5b80efa64f1f395c580cb8/dart/lib/src/hub.dart)? Then we can use it like
```dart
_hub.options.logger(SentryLevel.error, 'endTimestamp cannot be before startTimestamp');
```

Reply:
This is a question for @marandaneto, @denrase or @brustolin . It's probably useful because there are a couple other places where logging isn't available but where it could be useful.

Reply:
For cocoa we can access options from the client property in the hub. I don't see any problem with having and option getter in the hub, but Im also not aware of all implications that this could cause for Dart SDK.

Reply:
https://github.com/getsentry/sentry-dart/pull/716 is exposing the `options` via the `Hub` for internal-only usage.
The problem is that the options should be read-only, that's why it can't be public or misused.

Reply:
The PR is rebased to `main` and updated to use `_hub.options.logger` for this conversation.
I guess it's ready to review again @marandaneto, @denrase , @brustolin.